### DISCUSSION TITLE
how to customize build file locations

### DISCUSSION THREAD

**Discussion by firibz** [quasarframework/quasar#10309] [General - Components / Directives / etc]
Hello,
I need to change the default build file locations.
it is now like this:
![before](https://user-images.githubusercontent.com/7946986/128587546-3f2c8897-a164-47e1-aac2-436341dc8105.PNG)

but I need this:
![after](https://user-images.githubusercontent.com/7946986/128587571-08f447ae-7bc6-408f-9e33-264f429d3334.PNG)

**Comment by mfoitzik** [comment]
If I am reading this question correctly you want to go into your quasar.conf.js file and in the build: object add the following property: publicPath: 'everything',

The documentation for this is at: [https://quasar.dev/quasar-cli/quasar-conf-js#property-build](https://quasar.dev/quasar-cli/quasar-conf-js#property-build)

**Reply by firibz** [reply]
It is not what I am looking for. I need to have just 2 files in my dist folder.
1. index.html
2. a folder that contains everything except index.html

**Reply by ontwikkelfabriek** [reply]
@l3ibak 

don't think that's going to be possible without some serious hacking in Quasar's source code. I would ask some Quasar devs in on Discord for better answers.

**Reply by ecotrissy** [reply]
This seems to work (for the js and css):

quasar.conf.js

    build: {
	// ...

      chainWebpack (chain) {
        chain
          .output
          .filename('everything/[name].[chunkhash:8].js')
          .chunkFilename('everything/[name].[chunkhash:8].js')
        chain
          .plugin('mini-css-extract')
          .tap(args => [{ filename: 'everything/[name].[chunkhash:8].css' }, ...args])

        chain.plugin('eslint-webpack-plugin')
          .use(ESLintPlugin, [{ extensions: ['js', 'vue'] }])
      }

   },

**Comment by ecotrissy** [comment]
#Bump

I have a similar problem.
Server constraints are limiting me to a only single flat dist directory.
Is there any know solution please?

**Reply by ecotrissy** [reply]
What I've ended up doing, is writing a Python script to be run after building.
This flattens the directory hierarchy, then goes though all the files correcting the path references.
It's quite a horrible way to do things. But it does seem to work.

**Reply by MilosPaunovic** [reply]
After I read this, I got a bit curious myself so I've investigated a bit.

Haven't found the right solution to set up `Webpack` to bundle files flat, but may I suggest that you stay in the `Node` context and use `build` property in `quasar.conf.js`, and its [afterBuild hook](https://quasar.dev/quasar-cli/quasar-conf-js#:~:text=afterBuild(%7B%20quasarConf%20%7D)) to flatten the hierarchy and updated file references; might give you a cleaner solution.
